File: house-cleaning-backend/src/main/java/pl/nevernedingcode/service/cleaning/product/CleaningProductService.java
package pl.nevernedingcode.service.cleaning.product;

import org.springframework.http.ResponseEntity;
import pl.nevernedingcode.request.CLeaningProductPageRequest;
import pl.nevernedingcode.request.CleaningProductRequest;
import pl.nevernedingcode.response.CleaningProductResponse;
import pl.nevernedingcode.response.PageResponse;

public interface CleaningProductService {
    ResponseEntity<?> saveProduct(final CleaningProductRequest cleaningProductRequest);

    ResponseEntity<PageResponse<CleaningProductResponse>> getByProductName(final CLeaningProductPageRequest cLeaningProductPageRequest);
}

File: house-cleaning-frontend/src/app/cleaning-product/service/cleaning-product.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {PageResponse} from '../../core/model/page-response';
import {CleaningProductResponse} from '../model/cleaning-product-response';
import {CleaningProductRequest} from '../model/cleaning-product-request';
import {CLeaningProductPageRequest} from '../model/cLeaning-product-page-request';
import {AppConfig} from '../../config';

@Injectable({
  providedIn: 'root'
})
export class CleaningProductService {

  constructor(private http: HttpClient) {
  }

  getAllProducts(request: CLeaningProductPageRequest): Observable<PageResponse<CleaningProductResponse[]>> {
    return this.sendRequest('POST', `${AppConfig.serverAddress}/cleaning-product/get-by-product-name`, request);
  }

  saveProduct(request: CleaningProductRequest): Observable<PageResponse<any>> {
    return this.sendRequest('POST', `${AppConfig.serverAddress}/cleaning-product/save`, request);
  }

  private sendRequest<T, V>(httpMethod: string, url: string, body?: V): Observable<T> {
    return this.http.request<T>(httpMethod, url, {
      body: body
    });
  }

}
